### ISSUE TITLE
Reader should use React only

### ISSUE THREAD

**Issue by michael** [substance/journal#31]
That way we can use React.render on the server (React.renderToString). We need to find out if it's a problem that we use our own view implementation for rendering annotations. I guess when we just do that and use `dangerouslySetInnerHTML` we should be fine.

**Comment by Integral** [comment]
You want to prerender HTML on server?

**Comment by michael** [comment]
Yep it's very important so google can index that page.

**Comment by Integral** [comment]
aha... but maybe could do it without prerendered html? take a look:
http://googlewebmastercentral.blogspot.no/2014/05/understanding-web-pages-better.html

**Comment by Integral** [comment]
The thing is that in this case the server no more restfull and becoming multifunctional thing. In the other hand I'm not sure what is the state of indexing js apps.

**Comment by michael** [comment]
That's true, but i think that's okay. I thought of also serving the whole frontend as one single application, but that will be really hard to execute when we want to have decent SEO. So what i really care about is that the backend is just one react app, but for reader I think it has an advantage when we serve it a bit more classical. 

http://myjournal.com -> list of articles
http://myjournal.com/mydoc -> reader instance showing that article
http://myjournal.com/mydoc.epub (download epub for offline reading)

Regarding Google: I understand that it can index javascript-generated content but i know this is all quite limited. E.g. google doesn't click through multiple navigation paths... There's another more explicit option: the Ajax crawling scheme: https://support.google.com/webmasters/answer/174992?hl=en

Regarding React.js rendering on the server.. this is generally a nice thing.. because for instance on initial load of the page you already get the html, which removes the initial rendering delay. Jeh and then it's also crawlable.. it's not super high priority at the moment. but since we have a minimal setup now i want to investigate how well this all would work.

**Comment by Integral** [comment]
Ok, let's try! Anyway, it's very interesting. And also, no more super-large json files going to client, yes? In the other hand html would be larger...

**Comment by michael** [comment]
Mhh... we would still need the JSON, but we would serve that with the page so no extra http request is needed. I now realize though that this would mean the double amount of data sent e.g. for a large document, the HTML + the JSON.

**Comment by Integral** [comment]
Then we should do it only for google.

**Comment by michael** [comment]
Jeh maybe. Thing is that we will allow interactive actions on the doc, like revealing a comment and also writing a new comment. So we need to have a like a light-editor working (that supports annotating). And that can not be done with HTML only.
